lightningd/lightningd.c

```diff
@@ -182,6 +206,37 @@ static const char *find_my_path(const tal_t *ctx, const char *argv0)
 	tal_free(tmpctx);
 	return path_dirname(ctx, take(me));
 }
+static const char *find_my_pkglibexec_path(const tal_t *ctx,
+					   const char *my_path TAKES)
+{
+	char *relpath;
+	char *actualpath;
+	const tal_t *tmpctx = tal_tmpctx(ctx);
+
+	/* Compute relative path from bindir to pkglibexecdir. */
+	relpath = path_rel(tmpctx, BINDIR, PKGLIBEXECDIR);
+
+	/* Compute actual path, assuming my_path is BINDIR. */
+	actualpath = path_join(tmpctx, my_path, take(relpath));
+	relpath = NULL;
+
+	/* Simplify path. */
+	actualpath = path_simplify(ctx, take(actualpath));
+
+	tal_free(tmpctx);
+	return actualpath;
+}
+/* Determine the correct daemon dir. */
+static const char *find_daemon_dir(const tal_t *ctx, char const *argv0)
+{
+	const char *my_path = find_my_path(ctx, argv0);
+	if (has_all_daemons(my_path)) {
+		return my_path;
+	}
+	const char *pkglibexec_path
+		= find_my_pkglibexec_path(ctx, take(my_path));
+	return pkglibexec_path;
```

Comment:
How about just "return  find_my_pkglibexec_path(ctx, take(my_path));"?

Reply:
Yes, that would be better.